tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionParser.cs

```diff
@@ -0,0 +1,483 @@
+                                }
+
+                                return itParameter;
+                            }
+
+                            if (readerValue == "@exceptions")
+                            {
+                                return itParameter;
+                            }
+
+                            return Expression.Constant(readerValue);
+                        }
+
+                    case JsonToken.Integer:
+                        {
+                            return Expression.Constant(Convert.ChangeType(readerValue, TypeCode.Int32));
+                        }
+
+                    case JsonToken.StartArray:
+                        {
+                            _arrayStack++;
+                            break;
+                        }
+
+                    case JsonToken.EndArray:
+                        {
+                            _arrayStack--;
+                            break;
+                        }
+
+                    case JsonToken.Null:
+                        {
+                            return Expression.Constant(null);
+                        }
+                }
+            }
+            catch (Exception e)
+            {
+                AddError(reader.Value?.ToString() ?? "N/A", e.Message);
+                return ReturnDefaultValueExpression();
+            }
+        }
+
+        return null;
+    }
+
+    private void AddError(string expression, string error)
+    {
+        (_errors ??= new List<EvaluationError>()).Add(new EvaluationError { Expression = expression, Message = error });
+    }
+
+    private bool ConditionalRead(JsonTextReader reader, bool shouldAdvanceReader)
+    {
+        return !shouldAdvanceReader || reader.Read();
+    }
+
+    private void SetReaderAtExpressionStart(JsonTextReader reader)
+    {
+        while (reader.Read())
+        {
+            if (reader.TokenType != JsonToken.PropertyName)
+            {
+                continue;
+            }
+
+            if ((reader.Value?.ToString() == "json"))
+            {
+                reader.Read();
+                reader.Read();
+                return;
+            }
+        }
+
+        throw new InvalidOperationException("Invalid json file");
+    }
+
+    private Expression HandleReturnType(Expression finalExpr)
+    {
+        if (typeof(T).IsAssignableFrom(finalExpr.Type))
+        {
+            return finalExpr;
+        }
+
+        if (typeof(T) != typeof(string))
+        {
+            // let the caller throw the correct assign exception
+            return finalExpr;
+        }
+
+        // for string, call ToString or return the type name
+        if (SupportedTypesService.IsSafeToCallToString(finalExpr.Type))
+        {
+            finalExpr = Expression.Call(finalExpr, GetMethodByReflection(typeof(object), nameof(object.ToString), Type.EmptyTypes));
```

Comment:
Do you want to cache aside the `MethodInfo` that is returned from `GetMethodByReflection` of `Object.ToString`?
(It seems there is also a heavy utilization of `Object.ToString`'s `MethodInfo` in `ProbeExpressionParser.String.cs`)